### ISSUE TITLE
Adapter Xiaomi Air Purifier 3H

### ISSUE THREAD

**Issue by Haenk93** [ioBroker/AdapterRequests#385]
Es gibt bereits einen Adapter für den 2er und 2S Luftreiniger.
Leider ist der Ersteller wohl nicht mehr aktiv, da es seit einem Jahr keine Updates mehr gibt.
https://github.com/JoJ123/ioBroker.mihome-airpurifier

Wäre echt cool wenn den jemand updaten könnte für den neuen, dass ich ihn endlich über Iobroker ansteuern kann

**Comment by pur3asdf** [comment]
Würde mich auch sehr erfreuen

**Comment by maxpd1** [comment]
Same here. Wobie letzte Bewegung vor 3 Monaten war.

**Comment by elnowak** [comment]
Wäre ebenfalls interessiert

**Comment by MrLee99** [comment]
Seid heute auch interressiert :-)

**Comment by janhp** [comment]
push!

**Comment by yogiflop** [comment]
Na dann werde ich mich auch mal in die Liste der Nutzer eintragen. In der Hoffung, das sich mal jemand des Adapters annimmt.

**Comment by Pittini** [comment]
Ich bin da grad an nem Skript dran welches die [node-mihome](https://github.com/maxinminax/node-mihome) verwendet. Auslesen lässt sich das Ding schon, mit steuern haperts noch. Leider ist der Entwickler nicht der schnellste mit Issues beantworten, sonst gäbs zumindest schon mal ne Beta für Euch. Ich halte Euch auf dem laufenden wenn sich was tut.

**Comment by Pittini** [comment]
So, es wär soweit, ich bräuchte bitte Betatestwillige für das Skript. Siehe: https://github.com/Pittini/iobroker-nodemihome

**Comment by TiboTake** [comment]
Moin, ich war mal so frei dein Skript zu verwenden. Also einige Daten bekommt das Skript geliefert aber es ist leider nur sehr mager an aktuellen states oder werten. oder benötigt das Skript einige zeit um die Werte zu aktualisieren?

**Comment by Pittini** [comment]
Ja, ich weis, es hat sich leider rausgestellt dass ich in einem Punkt von einer falschen Grundannahme ausgegangen bin. Ich mach die Tage ein update, Du kannst das aber auch selber erstmal reparieren.
Wenn Du das Skript startest, sollten im Log Deine Geräte gelistet werden. Jetzt suchste den Eintrag vom Purifier und suchst dort nach der did. Jetzt gehste mit "suchen und ersetzen" über das Skript und lässt überall die 317335021 mit deiner did ersetzen. Jetzt läuft das Skript bis auf die beiden known issues, für die ich nix kann.

**Comment by TiboTake** [comment]
Läuft gut damit. Hab nun schonmal die Möglichkeit der Reiniger ein und aus zu schalten. Zwischen den Modis geht das schalten trotz dropdown menü noch nicht. Aber sonst ist deine arbeit echt top

**Comment by Pittini** [comment]
@TiboTake 
> Läuft gut damit. Hab nun schonmal die Möglichkeit der Reiniger ein und aus zu schalten. Zwischen den Modis geht das schalten trotz dropdown menü noch nicht. Aber sonst ist deine arbeit echt top

Jo, das steht ja auch in der readme, ich kann nix dafür wenn andere ihre PRs nicht mergen.
 Lösung dafür steht hier: https://github.com/Pittini/iobroker-nodemihome/issues/1

**Comment by Dugur** [comment]
Push!

**Comment by Crissii10** [comment]
Push! :)

**Comment by Pittini** [comment]
> Push! :)

Der Adapter scheint nicht weiterentwickelt zu werden, da helfen auch push's nix. Ich kann Euch nur meine Alternative anbieten, ist aber ein Skript und kein Adapter. Welche Geräte integriert sind, steht in der readme. https://github.com/Pittini/iobroker-nodemihome

**Comment by maxpd1** [comment]
und die Alternative ist ausreichend und gut :)

**Comment by Crissii10** [comment]
> und die Alternative ist ausreichend und gut :)

Nur sie funktioniert bei mir auf dem Pi nicht. Haben scheinbar mehrere das Problem.

**Comment by maxpd1** [comment]
was genau funktioniert nicht. Ab welcher Stelle der Anleitung klappt es nicht? Nutze ebenfalls den Pi.

**Comment by Crissii10** [comment]
> was genau funktioniert nicht. Ab welcher Stelle der Anleitung klappt es nicht? Nutze ebenfalls den Pi.

Die Anleitung an sich klappt. Aber es werden keine Geräte gefunden. Ich hab es irgendwann dann wieder entfernt. Ich hab noch ein Bild von log. Aber denke das ist der falsche Ort hier ![image](https://user-images.githubusercontent.com/84388739/118771804-41da8a80-b883-11eb-9967-75b6e0f24c93.png)

**Comment by maxpd1** [comment]
Ich habe die LoginDaten im Script hinterlegt und er hat mir automatisch alle meine Geräte als Datenpunkte angelegt.
Habe zudem alle zusätzlichen Devices als JS hinterlegt und er hat mir die Datenpunkte ab dem Zeitpunkt sauber befüllt. 
Vllt klappts mittlerweile mit der 0.2.15.

**Comment by Pittini** [comment]
Ja, es ist tatsächlich so dass es bei allen eigentlich sauber läuft und nur Pi Nutzer ein Problem haben. Aber auch nicht alle. Leider hab ich von Linux Systemebene nicht wirklich nen Plan und vom Pi schon 2mal nicht, ich selber nutze nen Nuc mit Proxmox und ner VM für den Iobroker. Evtl. kann sich ja mal wer ransetzen und rausfinden was da los ist, würde bei einigen Begeisterung auslösen.

**Comment by Pittini** [comment]
> Was bedeutet den waiting for pre merge?

Das der Dev der node-mihome meinen Pull Request merged, dann wär nämlich zumindest das Gedöns mit Files kopieren nimmer nötig. Leider scheint der schon seit Monaten kein Bock zu haben, da passiert gar nix.

**Comment by Wildbill-Z** [comment]
@Pittini 
> > Was bedeutet den waiting for pre merge?
> 
> Das der Dev der node-mihome meinen Pull Request merged, dann wär nämlich zumindest das Gedöns mit Files kopieren nimmer nötig. Leider scheint der schon seit Monaten kein Bock zu haben, da passiert gar nix.

Meinst Du, da tut sich noch irgendwann was?

**Comment by neuffer** [comment]
Hat es schon jemand geschafft einen Xiaomi Air Purifier 4 Pro in IOBroker zu bringen?
